File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        
        char[] arr = new char[128];
        int ans = 0;
        
        for(int i=0; i<s.length(); i++) {
            arr[s.charAt(i)]++;
        }
        for(int i=0; i<128; i++) {
            ans += (arr[i]/2); 
        }
        ans *= 2;
        if (ans < s.length()) {
            return ans+1;
        }
        return ans;

        
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        
        result = 0
        
        myMap = {}
        
        for c in s:
            if c not in myMap:
                myMap[c] = 0
            myMap[c] += 1
        
        for key in myMap.keys():
            result += myMap[key]//2
        
        result *= 2
        
        if len(s) > result:
            return result + 1
        else:
            return result
        